# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaTrackGraph.h
# Commit: a00ad7266f54
# Full Hash: a00ad7266f54820d7a3ff1233fa156f63190a3a3
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1651745
# File Overlap Count: 1
# Description:
#   Bug 1651745 - Signal AudioInputProcessing on input stream stop instead of driver start. r=padenot
#   
#   Without this patch, AudioInputProcessing wouldn't be aware of an audio driver
#   changing to another driver, be it a system driver or an audio driver that starts
#   on its fallback.
# ==============================================================================

diff -r a76d9158cdfd -r a00ad7266f54 dom/media/MediaTrackGraph.h
--- a/dom/media/MediaTrackGraph.h	Thu Nov 05 16:42:41 2020 +0000
+++ b/dom/media/MediaTrackGraph.h	Thu Nov 05 16:43:24 2020 +0000
@@ -116,10 +116,11 @@
                                 AudioDataValue* aBuffer, size_t aFrames,
                                 TrackRate aRate, uint32_t aChannels) = 0;
   /**
-   * An AudioCallbackDriver signaling that it has started and may notify of data
-   * soon.
+   * An AudioCallbackDriver with an input stream signaling that it has stopped
+   * for any reason and the AudioDataListener will not be notified of input data
+   * until the driver is restarted or another driver has started.
    */
-  virtual void NotifyStarted(MediaTrackGraphImpl* aGraph) = 0;
+  virtual void NotifyInputStopped(MediaTrackGraphImpl* aGraph) = 0;
   /**
    * Input data from a microphone (or other audio source.  This is not
    * guaranteed to be in any particular size chunks.